### PR TITLE
Add test fixtures to replace hard-coded values in tests and update InventoryStore unit tests

### PR DESCRIPTION
**Pull request by christine-w**
In addition to using fixtures instead of hardcoded parameters, the InventoryStore unit tests also had to be updated to handle the new shape of the db query results. (Context: The updated data object shape is a result of the query optimization in https://github.com/model-airbnb/Search/pull/25.)

### CODE REVIEW

File: test/InventoryStoreSpec.js [hunk 1]
```diff
@@ -1,129 +1,115 @@
...
+      db.getAvailableListings(TEST_MARKET, AVAILABLE_DATE_RANGE_START, AVAILABLE_DATE_RANGE_END, 1)
         .then((results) => {
-          results.forEach((result) => {
-            stayDates[result.inventory_date.toJSON().split('T')[0]] += 1;
-          });
-          Object.keys(stayDates).forEach((date) => {
-            expect(stayDates[date]).to.equal(1);
-          });
+          expect(results[0].dates.length).to.equal(AVAILABLE_DATE_RANGE_LENGTH);
+          expect(results.length).to.equal(1);
```
Comment (tylertruong) [inline test/InventoryStoreSpec.js:103]: Is this test still being performed as it was before? It looks different.

File: test/InventoryStoreSpec.js [hunk 2]
```diff
@@ -1,129 +1,115 @@
...
 
-    it('Should retrieve all 2 San Francisco listings available for all dates in the range 2017-11-10 to 2017-11-12 when top 100 is requested', (done) => {
-      const availableListings = new Set();
-      const stayDates = { '2017-11-10': 0, '2017-11-11': 0, '2017-11-12': 0 };
-      db.getAvailableListings('San Francisco', '2017-11-10', '2017-11-12', 100)
+    it(`Should retrieve all ${NUM_RANGE_AVAILABLE_LISTINGS} ${TEST_MARKET} listings available for all dates in the range ${AVAILABLE_DATE_RANGE_START} to ${AVAILABLE_DATE_RANGE_END} when top 100 is requested`, (done) => {
+      db.getAvailableListings(TEST_MARKET, AVAILABLE_DATE_RANGE_START, AVAILABLE_DATE_RANGE_END, 100)
         .then((results) => {
           results.forEach((result) => {
-            availableListings.add(result.listing_id);
```
Comment (tylertruong) [inline test/InventoryStoreSpec.js:121]: Shouldn't this be result => ( for Airbnb?

File: test/inventoryDb/fixtures.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+const AVAILABILITY_YEAR_MONTH = '2018-03';
+const SINGLE_AVAILABILITY_DAY = '26';
```
Comment (tylertruong) [inline test/inventoryDb/fixtures.js:2]: Are these supposed to be strings?

File: test/inventoryDb/fixtures.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+const NO_AVAILABILITY_DAY = '12';
+const AVAILABLE_RANGE_START_DAY = '19';
+const AVAILABLE_RANGE_END_DAY = '22';
+
+const availableDateRange = [];
+for (let d = parseInt(AVAILABLE_RANGE_START_DAY, 10);
+  d <= parseInt(AVAILABLE_RANGE_END_DAY, 10); d += 1) {
```
Comment (tylertruong) [inline test/inventoryDb/fixtures.js:9]: I don't understand this for loop?

File: test/inventoryDb/setup.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+const { database, pgConnection, testDbConnection } = require('./config');
+const { createTable, createIndex } = require('../../inventoryLocalStore/setup/queries');
+const {
+  LISTINGS_TABLE, listingAttributes, listings,
+  AVAILABILITY_TABLE, availabilityAttributes, availability,
+} = require('./fixtures');
+
+// Connect first to the postgres database to manage inventory_test database drop/create
+let pool = new Pool({
+  connectionString: pgConnection,
```
Comment (tylertruong) [inline test/inventoryDb/setup.js:14]: const pool.

**Review by tylertruong — APPROVED** [review]
Great unit tests! These are just suggestions!

File: test/InventoryStoreSpec.js (hunk 1 above)
Comment (christine-w) [reply test/InventoryStoreSpec.js:103]: Yup! The db result sets used to return a row for every availability date so a three-night stay with one available listing would result in three rows. Now, the db result set returns one row per listing that is available for all dates in the range and the available nights are in an array in the dates column. So to confirm that the listings returned are available for the entire duration, the test now only needs to check the length of the dates array.

File: test/InventoryStoreSpec.js (hunk 2 above)
Comment (christine-w) [reply test/InventoryStoreSpec.js:121]: No, there is no return value following it.

File: test/inventoryDb/fixtures.js (hunk 4 above)
Comment (christine-w) [reply test/inventoryDb/fixtures.js:9]: Airbnb linting complained about the length of this line, hence the broken up for loop. :( But, this is getting changed in an upcoming pull request and the line length will be short enough to fit back onto one line.

File: test/inventoryDb/fixtures.js (hunk 3 above)
Comment (christine-w) [reply test/inventoryDb/fixtures.js:2]: Yes, these are intended for easy construction of date strings. Having said that, the date values will be updated to numbers in an upcoming pull request due to the need to do math with them.

File: test/inventoryDb/setup.js (hunk 5 above)
Comment (christine-w) [reply test/inventoryDb/setup.js:14]: No, pool gets reassigned further down so it can't be a const.

**Pull request merged**